docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
+|v1.0|user|remove items|remove the items that are no longer required to be tracked|
+|v1.0|user|add new orders|track the orders added to the system|
+|v1.0|user|view the orders in the system|track the orders that are added or not added to the system|
+|v1.0|user|remove orders|remove the orders that are no longer required to be tracked|
+|v1.0|user|clear all items|re-track all the items if there are any problems|
+|v1.0|user|clear all orders|re-track all the orders if there are any problems|
+|v1.0|user|view all available commands|refer to them when I forget how to use the application|
+|v1.0|user|view all items-related commands|refer to them when I forget how to use the items feature|
+|v1.0|user|view all orders-related commands|refer to them when I forget how to use the orders feature|
+|v2.0|user|create, save data and load existing data|work on another device with the saved data|
+|v2.0|expert user|create, save data and load existing data|edit the data file directly|
+|v2.0|user|delete multiple items|save time by not deleting items one by one|
+|v2.0|user|delete multiple orders|save time by not deleting orders one by one|
+|v2.0|user|find relevant items|save time by looking through relevant items|
+|v2.0|user|find relevant orders|save time by looking through relevant orders|
+|v2.0|user|add a new item with a specified unit price|calculate order price more easily|
+|v2.0|user|update the unit price of an item|revise unit price or correct mistaken unit price entered earlier|
+|v2.0|user|add a new item with its stock|keep track of total number of stock and calculate order price more easily|
+|v2.0|user|update the stock of an item|Correct mistaken stock number entered earlier|
+|v2.0|user|specify the quantity of a item to add to the order|calculate the total price of the order and update the item inventory correctly.|
+|v2.0|user|check the total price of a specific order|feedback to customer when they ask for it and do not have to look through the order list.|
+|v2.0|user|update the status of an order|differentiate the orders by their status.|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+- Be able to work on any mainstream OS as long as it has Java 11 or above installed.
+- Be able to respond to any command in less than 2 seconds.
+- A user should be able to complete majority of tasks faster using CLI than GUI.
 
 ## Glossary
 
-* *glossary item* - Definition
+Abbreviation | Full title | Definition
+-------- | ---------- | ----------
+**CI**   | Continuous Integration | Combining parts of a software product to form a whole
+**IntelliJ** | IntelliJ | An Integrated Development Environment written in Java
+**CLI** | Command Line Interface | A program that accepts text inputs to execute operating system functions
+**GUI** | Graphical User Interface | An interface that allows users to interact through graphical icons
+**Mainstream OS** | Windows, Linux, Unix, OS-X | Operating systems
+
```

Comment:
I like the inclusion of a glossary of technical terms to help the people who are unfamiliar with these terms.